### ISSUE TITLE
Backslash escapes (\`) is lost in template literal strings

### ISSUE THREAD

**Issue by AndersMad** [trullock/NUglify#245]
Problem: The backslash escape char is removed

This:
```js
NUglify.Uglify.Js(@"let s = `a is \`${a}\`...`;").Code
```
Translates into:
```js
let s=`a is `${a}`...`
```
Should be:
```js
let s=`a is \`${a}\`...`
```

**Comment by trullock** [comment]
Thanks for reporting.

Are you on the latest version?

There were some escaping fixes made recently....

**Comment by AndersMad** [comment]
Hi, sry forgot to write - yes: **v1.13.8** via nuget

**Comment by trullock** [comment]
Ok see the recent releases and commits for clues to where the escaping bugs lie. 

Otherwise I'll take a look asap

**Comment by trullock** [comment]
The issue lies in `JSParser` `ParseTemplateLiteral`.

It assumes there will never be escaped backticks and so treats them as regular characters

Its currently unclear why `ParseTemplateLiteral` seems to be expected to be called on "partial" template literals, i.e. ones that do/dont start/end with backticks. Need to understand this before this can be fixed, otherwise template literals which start (or more likely end) with an escaped backtick might not be handled properly

**Comment by trullock** [comment]
Hopefully fixed in 1.17.3
